data-processing/common/parse_tex.py

```diff
@@ -617,6 +648,44 @@ def parse_soup(tex: str) -> TexSoup:
     except (TypeError, EOFError) as e:
         raise TexSoupParseError(str(e))
 
+def reserved_char_check(tex):
```

Comment:
Looks great! One small nit pick: can these function be renamed to be an action rather than a noun (e.g., `check_for_reserved_characters`)